.travis.yml

```diff
@@ -1,20 +1,44 @@
 language: ruby
 rvm:
   - 2.3.1
+
 branches:
   only:
     - master
     - develop
-script: bundle exec rspec --tag ~broken_in_ci
+
+services:
+  - mysql
+  - postgresql
+
+env:
+  - DB=sqlite
+  - DB=mysql
+  - DB=postgres
+
+before_install:
+  - if [[ $DB = 'sqlite' ]]; then echo "gem 'sqlite3'" > Gemfile.local; fi
+  - if [[ $DB = 'mysql' ]]; then echo "gem 'mysql2'" > Gemfile.local; fi
+  - if [[ $DB = 'postgres' ]]; then echo "gem 'pg'" > Gemfile.local; fi
+
 install:
   - bundle install --jobs=3 --retry=3
-  - bundle exec sqlite3 --version
+
+before_script:
   - bundle exec rake config
+  - cp config/database.yml.travis-$DB config/database.yml
+  - if [[ $DB = 'sqlite' ]]; then bundle exec rake db:create; fi
+  - if [[ $DB = 'mysql' ]]; then mysql -e 'CREATE DATABASE dpn;'; fi
+  - if [[ $DB = 'postgres' ]]; then psql -c 'CREATE DATABASE dpn;' -U postgres; fi
+  - bundle exec rake db:migrate
+
+script: bundle exec rspec --tag ~broken_in_ci
+
```

Comment:
I want to be very careful about the complexity in the ci server leaking into the codebase.  Having the extra database.yml files was a necessary evil I couldn't manage to workaround.  So restricting `broken_in_ci` to `broken_in_sqlite` is great, but I don't want to one day also have `broken_in_mysql`, `broken_in_postgres`, `broken_in_mongo`, `broken_on_tuesdays`, `broken_in_linux`, etc.